cloudpathlib/cloudpath.py

```diff
@@ -56,21 +56,31 @@
 else:
     from typing_extensions import Self
 
-if sys.version_info >= (3, 12):
+if sys.version_info[:2] == (3, 12):
     from pathlib import posixpath as _posix_flavour  # type: ignore[attr-defined]
     from pathlib import _make_selector  # type: ignore[attr-defined]
-else:
-    from pathlib import _posix_flavour  # type: ignore[attr-defined]
-    from pathlib import _make_selector as _make_selector_pathlib  # type: ignore[attr-defined]
 
-    def _make_selector(pattern_parts, _flavour, case_sensitive=True):
+if sys.version_info < (3, 12):
+    from pathlib import _posix_flavour  # type: ignore[attr-defined] # noqa: F811
+    from pathlib import _make_selector as _make_selector_pathlib  # type: ignore[attr-defined] # noqa: F811
+
+    def _make_selector(pattern_parts, _flavour, case_sensitive=True):  # noqa: F811
         return _make_selector_pathlib(tuple(pattern_parts), _flavour)
 
 
+if sys.version_info >= (3, 13):
+    from pathlib._local import _PathParents
+    import posixpath as _posix_flavour  # type: ignore[attr-defined]   # noqa: F811
+
+    from .legacy.glob import _make_selector  # noqa: F811
+else:
+    from pathlib import _PathParents  # type: ignore[attr-defined]
```

Comment:
Nit, but this feels kind of confusingly arranged right now. I think this would be more readable if it were something like

```
if sys.version_info < (3,12):
    ...
elif sys.version_info[:2] == (3,12):
    ...
else: 
    ...

where it's in either ascended or descending order, and all of the imports for each case is consolidated.